actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -695,6 +695,26 @@ def transform_keys!(&block)
       self
     end
 
+    # Returns a new <tt>ActionController::Parameters</tt> instance with the
+    # results of running +block+ once for every key. This includes the keys
+    # from the root hash and from all nested hashes. The values are unchanged.
+    def deep_transform_keys(&block)
+      if block
+        new_instance_with_inherited_permitted_status(
+          @parameters.deep_transform_keys(&block)
+        )
+      else
+        @parameters.deep_transform_keys
```

Comment:
It looks like this was copied from [the `transform_keys` implementation](https://github.com/rails/rails/blob/aa7da0471f7fcf2adbbc26c37a25914d8493928d/actionpack/lib/action_controller/metal/strong_parameters.rb#L687). It makes sense there, because calling `transform_keys` on a hash without passing a block returns an enumerator:

```ruby
irb(main):001:0> { foo: "bar" }.transform_keys
=> #<Enumerator: {:foo=>"bar"}:transform_keys>
```

However that doesn't work for `deep_transform_keys`:

```ruby
irb(main):002:0> { foo: "bar" }.deep_transform_keys
Traceback (most recent call last):
        7: from tools/console:11:in `<main>'
        6: from (irb):2
        5: from /Users/eugene/src/rails/activesupport/lib/active_support/core_ext/hash/keys.rb:66:in `deep_transform_keys'
        4: from /Users/eugene/src/rails/activesupport/lib/active_support/core_ext/hash/keys.rb:119:in `_deep_transform_keys_in_object'
        3: from /Users/eugene/src/rails/activesupport/lib/active_support/core_ext/hash/keys.rb:119:in `each_with_object'
        2: from /Users/eugene/src/rails/activesupport/lib/active_support/core_ext/hash/keys.rb:119:in `each'
        1: from /Users/eugene/src/rails/activesupport/lib/active_support/core_ext/hash/keys.rb:120:in `block in _deep_transform_keys_in_object'
LocalJumpError (no block given (yield))
```

Which means this method's implementation could be simplified:

```ruby
    def deep_transform_keys(&block)
      new_instance_with_inherited_permitted_status(
        @parameters.deep_transform_keys(&block)
      )
    end
```

Reply:
Thank you for the feedback. I pushed a commit with your suggestion.

Reply:
Thank you for your feedback. Pushed a commit with your suggestion.